src/client/index.ts

```diff
@@ -18,111 +79,268 @@ export interface NotificationOptions {
-    buttonIndex: number;
 }
 
+export function addEventListener(eventType: 'notification-action', listener: (event: NotificationActionEvent) => void): void;
+export function addEventListener(eventType: 'notification-created', listener: (event: NotificationCreatedEvent) => void): void;
+export function addEventListener(eventType: 'notification-closed', listener: (event: NotificationClosedEvent) => void): void;
+
 /**
- * @hidden
+ * Adds a listener, see definitions of individual event interfaces for details on each event.
+ *
+ * @param eventType The event being subscribed to
+ * @param listener The callback function to add
  */
-export type NotificationEvent = NotificationClickedEvent | NotificationClosedEvent | NotificationButtonClickedEvent;
-
-export function addEventListener(eventType: 'notification-clicked', listener: (event: NotificationClickedEvent) => void): void;
-export function addEventListener(eventType: 'notification-closed', listener: (event: NotificationClosedEvent) => void): void;
-export function addEventListener(eventType: 'notification-button-clicked', listener: (event: NotificationButtonClickedEvent) => void): void;
-export function addEventListener<E extends NotificationEvent>(eventType: E['type'], listener: (event: E) => void): void;
-export function addEventListener<E extends NotificationEvent>(eventType: E['type'], listener: (event: E) => void): void {
+export function addEventListener<E extends Events>(eventType: E['type'], listener: (event: E) => void): void {
     if (typeof fin === 'undefined') {
         throw new Error('fin is not defined. The openfin-notifications module is only intended for use in an OpenFin application.');
     }
 
     eventEmitter.addListener(eventType, listener);
 }
 
-export function removeEventListener(eventType: 'notification-clicked', listener: (event: NotificationClickedEvent) => void): void;
+export function removeEventListener(eventType: 'notification-action', listener: (event: NotificationActionEvent) => void): void;
+export function removeEventListener(eventType: 'notification-created', listener: (event: NotificationCreatedEvent) => void): void;
 export function removeEventListener(eventType: 'notification-closed', listener: (event: NotificationClosedEvent) => void): void;
-export function removeEventListener(eventType: 'notification-button-clicked', listener: (event: NotificationButtonClickedEvent) => void): void;
-export function removeEventListener<E extends NotificationEvent>(eventType: E['type'], listener: (event: E) => void): void;
-export function removeEventListener<E extends NotificationEvent>(eventType: E['type'], listener: (event: E) => void): void {
+
+/**
+ * Removes a listener previously added with {@link addEventListener}.
+ *
+ * Has no effect if `eventType` isn't a valid event, or `listener` isn't a callback registered against `eventType`.
+ *
+ * @param eventType The event being unsubscribed from
+ * @param listener The callback function to remove
```

Comment:
Standard behaviour for event emitters, but may be worth saying explicitly that this uses `"==="` for the listener